rfcs/rework-index-updates.md

```diff
@@ -0,0 +1,60 @@
+# Summary
+
+This RFC proposes moving away from the current model of fetching new releases
+to build, moving from running [crates-index-diff] in a thread to using webhooks
+and the [crates-index] crate.
+
+# Motivation
+
+While the current approach has worked well for us so far, it has some problems:
+
+* Running the update in a cronjob every 2 minutes is wasteful, as there is
+  often a greater delay between two publishes.
+* Running the update in a crobjob every 2 minutes adds delay to getting the
+  documentation built if the queue is empty, as the release might potentially
+  have to wait those extra two minutes.
+* The approach doesn't scale, if we want to move to a setup where there is more
+  than a single frontend server we'd have to elect which server runs the fetch.
+* The way crates-index-diff stores its state (a branch in the local repo) is
+  fragile, as it might become out of sync causing the loss of a publish.
+* The way crates-index-diff stores its state makes it hard to move the server
+  installation, as the index repository needs to be moved as well.
+
+# Proposal
+
+We configure the `crates.io-index` repository to send a webhook to a new
+endpoint, `/_/index-webhook`, which starts a index sync in the background. The
+payload of the webhook is ignored, but the webhook signature is validated if a
+secret key is provided to the application through an environment variable.
```

Comment:
Not sure I quite follow the bit about secret keys, is this like a public/private key or a symmetric key? Also, I assume that if no secret key is provided, it fails open so you can still update the index while developing locally?

Reply:
It's a thing with github, it shouldn't really affect local development. You give GH a key, they hash the content of their `POST` with that key and then you verify that the key used for hashing matches the one you hold

Reply:
Yep, that's to avoid people spamming that URL manually.